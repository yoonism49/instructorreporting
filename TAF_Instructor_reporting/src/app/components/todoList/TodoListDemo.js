// @flow weak

import React              from 'react';
import TodoList           from './todoList/TodoList';
import TodoListItem       from './todoListItem/TodoListItem';
import TodoListCommands   from './todoListCommands/TodoListCommands';
import TodoListAddTask    from './todoListCommands/TodoListAddTask';
import TodoListSeeAllTask from './todoListCommands/TodoListSeeAllTask';
import Panel              from '../panel/Panel';

const TodoListDemo = () => (
  <Panel
    hasTitle={true}
    title={'Todo list'}
    sectionCustomClass="tasks-widget">
    <TodoList>
      <TodoListItem
        label="Director is Modern Dashboard"
        done={false}
        statusLabel="2 days"
        statusLabelStyle="label-success"
      />
      <TodoListItem
        label="Fully Responsive & Bootstrap 3.0.2 Compatible"
        done={false}
        statusLabel="Done"
        statusLabelStyle="label-danger"
      />
      <TodoListItem
        label="Latest Design Concept"
        done={false}
        statusLabel="Company"
        statusLabelStyle="label-warning"
      />
      <TodoListItem
        label="Write well documentation for this theme"
        done={false}
        statusLabel="3 Days"
        statusLabelStyle="label-primary"
      />
      <TodoListItem
        label="Don't bother to download this Dashbord"
        done={false}
        statusLabel="Now"
        statusLabelStyle="label-inverse"
      />
      <TodoListItem
        label="Give feedback for the template"
        done={false}
        statusLabel="2 Days"
        statusLabelStyle="label-success"
      />
      <TodoListItem
        label="Tell your friends about this admin template"
        done={false}
        statusLabel="Now"
        statusLabelStyle="label-danger"
      />
    </TodoList>
    <TodoListCommands>
      <TodoListAddTask />
      <TodoListSeeAllTask />
    </TodoListCommands>
  </Panel>
);

export default TodoListDemo;
