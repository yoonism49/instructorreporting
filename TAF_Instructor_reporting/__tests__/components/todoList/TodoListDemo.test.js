
'use strict';

import React                    from 'react';
import TodoListDemo             from '../../../src/app/components/todoList/TodoListDemo';
import renderer                 from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
// import { mount }          from 'enzyme';

describe('TodoListDemo component', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <TodoListDemo />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
