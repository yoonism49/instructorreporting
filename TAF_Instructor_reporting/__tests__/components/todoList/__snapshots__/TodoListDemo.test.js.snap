// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListDemo component renders as expected 1`] = `
<div>
  <section
    className="panel tasks-widget"
  >
    <header
      className="panel-heading"
    >
      Todo list
    </header>
    <div
      className="panel-body undefined"
      style={
        Object {
          "backgroundColor": "#FFF",
        }
      }
    >
      <div
        className="task-content"
      >
        <ul
          className="task-list"
        >
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Director is Modern Dashboard
              </span>
              <span
                className="label label-success"
              >
                2 days
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Fully Responsive & Bootstrap 3.0.2 Compatible
              </span>
              <span
                className="label label-danger"
              >
                Done
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Latest Design Concept
              </span>
              <span
                className="label label-warning"
              >
                Company
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Write well documentation for this theme
              </span>
              <span
                className="label label-primary"
              >
                3 Days
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Don't bother to download this Dashbord
              </span>
              <span
                className="label label-inverse"
              >
                Now
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Give feedback for the template
              </span>
              <span
                className="label label-success"
              >
                2 Days
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
          <li>
            <div
              className="task-checkbox"
            >
              <input
                checked={false}
                className="flat-grey list-child"
                type="checkbox"
              />
            </div>
            <div
              className="task-title"
            >
              <span
                className="task-title-sp"
              >
                Tell your friends about this admin template
              </span>
              <span
                className="label label-danger"
              >
                Now
              </span>
              <div
                className="pull-right hidden-phone"
              >
                <button
                  className="btn btn-default btn-xs"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-pencil"
                  />
                </button>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        className=" add-task-row"
      >
        <a
          className="btn btn-success btn-sm pull-left"
          href="#"
        >
          Add New Tasks
        </a>
        <a
          className="btn btn-default btn-sm pull-right"
          href="#"
        >
          See All Tasks
        </a>
      </div>
    </div>
  </section>
</div>
`;
